File: 2º/PDOO/ProyectoRUBY/lib/c_juez.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "casilla.rb"
require_relative "jugador.rb"

module Civitas
  class C_juez<Casilla
    def initialize(numCasillaCarcel, nombre)
      super(nombre)
      @numCasillaCarcel=numCasillaCarcel
    end
    def  to_s
      devolver=super()
      devolver+="\n}\n"
      return devolver;
    end 
    def recibeJugador(actual, todos)
      if(actual<=todos.size() && actual>0)then
        super.informe(actual, todos)
        todos[actual].encarcelar(@carcel)
      end
    end

  end
end

File: 2º/PDOO/ProyectoJAVA/src/civitas/C_juez.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author nikitastetskiy
 */
public class C_juez extends Casilla{
    
    private static int carcel;
    
    C_juez(int numCasillaCarcel, String nombre){       // juez
        super(nombre);
        carcel = numCasillaCarcel;
    }
    
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).encarcelar(carcel);
        }
    }
    
    @Override
    public String toString() {
        String e = "\n-> Casilla ("+ super.getNombre() + ")";
        return e;
    }
}
